### DISCUSSION TITLE
Camera in "free" mode

### DISCUSSION THREAD

**Discussion by lufermar** [slmsuite/slmsuite#44] [Q&A] [answered]
Hello all,
I am using a Thorlab cs165 camera which I managed to connect to python via this package. 
I can take individual pictures and plot them for example, change hardware properties and also change the capture mode to "free" mode.
However I don't know how to show some sort of live video of what the camera is capturing in this mode. Or at least to show some numerical value that changes when the amount of light going to the camera changes. 
I tried `is_capturing()` and I get **True**. 
I do not expect to obtain something like the Thorlabs interface, but some live indicator would be nice and I do not seem to find it. 
So my question is: is it possible to do it with this package? If so, could you give me some indications?

Best,
L.

**Comment by ichristen** [comment, marked as answer]
Hi @lufermar ,

Thanks for asking about this; it is an area that we intend to improve in the future.

- Unfortunately, the long-term pyglet implementation of a live video view on the `pygletcamera` branch is unfinished. 
- There, however, is a matplotlib interface (currently private because it will eventually be replaced) in `slmsuite.hardware.cameras._view_continuous()` [here](https://github.com/QPG-MIT/slmsuite/blob/a977cfcf5eaf79d4fc702ef23a617c961fa83f79/slmsuite/hardware/cameras/camera.py#L446).
- Alternatively, you can `.close()` the camera in Python and open it in ThorCam or your live GUI of choice, then reopen the camera in Python after you're done.

`is_capturing()` refers to whether the `'free'` mode is active and the camera is gathering frames, in line with the Thorlabs SDK definition.
